### PR TITLE
Add basic 3d support

### LINKED ISSUE #32: Add 3D support
* 3D models
* PBR?
* support multiple render pipelines?

### PR DESCRIPTION
**Pull request by KyaZero**
Partially closes #32 

Adds simple 3D support in our pipeline, which should be somewhat easy to expand upon from now on!

### CODE REVIEW

File: editor/src/Panels/Scene/SceneHierarchyPanel.h [hunk 1]
```diff
@@ -23,6 +23,8 @@ namespace Wraith
         void OnUIRender() override;
         const char* GetName() const override { return "Scene Hierarchy"; }
 
+        Entity GetSelectedEntity() { return m_SelectedEntity; }
+
```
Comment (Husenap) [inline editor/src/Panels/Scene/SceneHierarchyPanel.h:27]: This is never used

File: engine/src/Core/Math/Vec3.h [hunk 2]
```diff
@@ -93,6 +93,7 @@ namespace Wraith
         inline Vec3 operator/(const Vec3& other) const { return { x / other.x, y / other.y, z / other.z }; }
         inline Vec3 operator*(const T& scalar) const { return { x * scalar, y * scalar, z * scalar }; }
         inline Vec3 operator/(const T& scalar) const { return { x / scalar, y / scalar, z / scalar }; }
+        inline Vec3 operator-() const { return { x * -1.0f, y * -1.0f, z * -1.0f }; }
```
Comment (Husenap) [inline engine/src/Core/Math/Vec3.h:96]:
```suggestion
        inline Vec3 operator-() const { return { -x, -y, -z }; }
```

File: engine/src/Graphics/Models/ModelManager.cpp [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        vertices.push_back({ Vec4f(-0.5f, -0.5f, 0.5f, 1.0f),  Vec4f(0.0f, 0.0f, 1.0f, 0.0f),  Vec4f(0.0f, 0.0f, 0.0f, 0.0f), Vec4f(0.0f, 0.0f, 0.0f, 0.0f), Vec4f(1.0f, 1.0f, 1.0f, 1.0f), Vec2f(1.0f, 0.0f) });
+                                 
+        vertices.push_back({ Vec4f(0.5f,  0.5f, -0.5f, 1.0f),  Vec4f(0.0f, 0.0f, -1.0f, 0.0f), Vec4f(0.0f, 0.0f, 0.0f, 0.0f), Vec4f(0.0f, 0.0f, 0.0f, 0.0f), Vec4f(1.0f, 1.0f, 1.0f, 1.0f), Vec2f(1.0f, 1.0f) });
+        vertices.push_back({ Vec4f(-0.5f,  0.5f, -0.5f, 1.0f), Vec4f(0.0f, 0.0f, -1.0f, 0.0f), Vec4f(0.0f, 0.0f, 0.0f, 0.0f), Vec4f(0.0f, 0.0f, 0.0f, 0.0f), Vec4f(1.0f, 1.0f, 1.0f, 1.0f), Vec2f(0.0f, 1.0f) });
+        vertices.push_back({ Vec4f(-0.5f, -0.5f, -0.5f, 1.0f), Vec4f(0.0f, 0.0f, -1.0f, 0.0f), Vec4f(0.0f, 0.0f, 0.0f, 0.0f), Vec4f(0.0f, 0.0f, 0.0f, 0.0f), Vec4f(1.0f, 1.0f, 1.0f, 1.0f), Vec2f(0.0f, 0.0f) });
+        vertices.push_back({ Vec4f(0.5f, -0.5f, -0.5f, 1.0f),  Vec4f(0.0f, 0.0f, -1.0f, 0.0f), Vec4f(0.0f, 0.0f, 0.0f, 0.0f), Vec4f(0.0f, 0.0f, 0.0f, 0.0f), Vec4f(1.0f, 1.0f, 1.0f, 1.0f), Vec2f(1.0f, 0.0f) });
```
Comment (Husenap) [inline engine/src/Graphics/Models/ModelManager.cpp:55]: would be nice to add constant variables for tangent, bitangent and color

**Review by Husenap — CHANGES_REQUESTED** [review]
Looks pretty good 🚀

**Review by Husenap — APPROVED** [review]

File: engine/src/Graphics/Models/ModelManager.cpp (hunk 3 above)
Comment (KyaZero) [reply engine/src/Graphics/Models/ModelManager.cpp:55]: yep!

File: engine/src/Core/Math/Vec3.h (hunk 2 above)
Comment (KyaZero) [reply engine/src/Core/Math/Vec3.h:96]: small brain me, thanks!

File: editor/src/Panels/Scene/SceneHierarchyPanel.h (hunk 1 above)
Comment (KyaZero) [reply editor/src/Panels/Scene/SceneHierarchyPanel.h]: Fixed!

File: editor/src/Editor.h [hunk 4]
```diff
@@ -2,12 +2,10 @@
...
 #include <memory>
 
-#include <Application.h>
-#include <Scene/Entity.h>
-#include <Scene/Scene.h>
-
 #include "Panels/PanelGroup.h"
 #include "Panels/Play/ViewportPanel.h"
+#include "Scene/EditorCamera.h"
+#include "Wraith.h"
```
[the remark this answers is not in the document]
Comment (KyaZero) [reply editor/src/Editor.h]: Fixed!

**Pull request merged**